Docu: updated Input-doc to refer to Refinery and current SotA (#5318)
ILIAS-eLearning/ILIAS: docs/development/input-processing.md

mutators. The integrity needs to be enforced by the datatype itself to make invalid
data unexpressable.

### Refinery

> The [Refinery library](../../src/Refinery) defines two basic abstractions to be used

Reviewer: ```suggestion
The [Refinery-library](../../src/Refinery) defines two basic abstractions to be used
```

for processing input securely:

* A **[`Transformation`](../../src/Refinery/Transformation.php)** is a structural

>   conversion between different shapes and types of data. It does not perform

Reviewer: From where I stand "perform" sounds a little strange in this context. I think "cause" side effects may be more adequate.

Reviewer 2: +1

  side effects. `Transformation`s can be combined to create complex `Transformation`s

>   from simpler ones. The [Refinery](../../Refinery) aims providing common

Reviewer 2: ```suggestion
  from simpler ones. The [Refinery](../../Refinery) aims to provide common
```

>   `Transformations` and combinators for them to be reused throughout the system.

Reviewer: ```suggestion
  `Transformation`s and combinators for them to be reused throughout the system.
```

>   `Transformations` and combinators for them to be reused throughout the system.

Reviewer 2: ```suggestion
  `Transformation`s and combinators so that they can be reused throughout the system.
```

  It thus is the tool to transform primitive input data into semantically richer
  structures in a declarative way.
* A **[`Constraint`](../../src/Refinery/Constraint.php)** is a check on some data in

>   conjunction with a builder for a human readable error-message. Like `Transformation`s,

Reviewer 2: ```suggestion
  conjunction with a builder for a human-readable error-message. Like `Transformation`s,
```

will also require additions to the [Data-library](../../src/Data) and the components
that use their own data structures.

> The [Refinery-library](../../src/Refinery) also offers a perspective how other policy

Reviewer: I think this should be "...a perspective on how other components that enforce policies..."

Reviewer 2: +1
